data/org.freedesktop.portal.ProxyResolver.xml

```diff
@@ -41,11 +46,53 @@
         proxy uri are of the form ``protocol://[user[:password] AT host:port``.
         The protocol can be ``http``, ``rtsp``, ``socks`` or another proxying
         protocol. ``direct://`` is used when no proxy is needed.
+
+        This is a convenience wrapper over reading the PAC file or the other
+        portal settings, in that order of priority from highest to lowest.
     -->
     <method name="Lookup">
       <arg type="s" name="uri" direction="in"/>
       <arg type="as" name="proxies" direction="out"/>
     </method>
+    <!--
+    The following values are valid:
+      0: Proxy should not be used.
+      1: Automatic proxy mode; use pac_url.
+      2: Manual proxy mode; read associated settings manually.
+
+      This property was added in version 2.
+    -->
+    <property name="use_proxy" type="u" access="read"/>
+    <!--
+      This property was added in version 2.
+    -->
+    <property name="pac_url" type="s" access="read"/>
+    <!--
+      This property was added in version 2.
+    -->
+    <property name="https_proxy" type="as" access="read"/>
+    <!--
+      This property was added in version 2.
+    -->
+    <property name="http_proxy" type="as" access="read"/>
+    <!--
+      socks_proxy:
+      An integer value of 0 is equivalent to SOCKS; of 4 is SOCKS4,
+      and 5 is SOCKS5. Any other value is invalid, but if present, can
+      be assumed to be equivalent to 0.
+
+      This property was added in version 2.
+    -->
+    <property name="socks_proxy" type="a(su)" access="read"/>
+    <!--
+      This property was added in version 2.
+    -->
+    <property name="ftp_proxy" type="as" access="read"/>
+    <!--
+      This property was added in version 2.
+    -->
+    <property name="ignored_hosts" type="as" access="read"/>
     <property name="version" type="u" access="read"/>
   </interface>
 </node>
+
```

Comment:
I strongly suspect some bug is Builder is causing it to sometimes add blank lines at the ends of files; perhaps there is some bug in its code to ensure the file ends in a newline character, which is actually useful. We shouldn't be committing blank lines at end of file.